api/src/main/java/com/github/eyefloaters/console/api/model/Node.java

```diff
@@ -4,10 +4,47 @@
 import com.fasterxml.jackson.annotation.JsonInclude.Include;
 
 @JsonInclude(value = Include.NON_NULL)
-public record Node(int id, String host, int port, String rack) {
+public class Node {
+
+    private final int id;
+    private final String host;
+    private final int port;
+    private final String rack;
+    private Either<ReplicaInfo, Error> log;
+
+    public Node(int id, String host, int port, String rack) {
+        this.id = id;
+        this.host = host;
+        this.port = port;
+        this.rack = rack;
+        this.log = null;
+    }
 
     public static Node fromKafkaModel(org.apache.kafka.common.Node node) {
         return new Node(node.id(), node.host(), node.port(), node.rack());
     }
 
+    public int getId() {
```

Comment:
I originally left them without the `get` then found that Jackson doesn't auto-serialize them. It's nothing some annotations can't fix. I don't oppose the record-like naming for mutators/accessors and it would give us consistency between the class-based and record-based models. My preference would be to change it for all the models in a separate PR without any functional changes.

Reply:
Sounds good. While needing annotations in the class declaration is annoying, I think consistency at call sites (of which there will be many many more) is more important.

Reply:
I reverted the `Node` changes and added `PartitionReplica` to hold a replica's `nodeId`, `nodeRack`, `inSync` status , and `localStorage`. I suspect we would have a `remoteStorage` field in the future with tiered storage information, but of course we need to wait for that to be available.

Reply:
Opened #43 to remember to go back and use the record-style accessors throughout the model.